pyomo/contrib/pyros/uncertainty_sets.py

```diff
@@ -765,33 +765,60 @@ def parameter_bounds(self):
         parameter_bounds = [(nom_value[i] - half_length[i], nom_value[i] + half_length[i]) for i in range(len(nom_value))]
         return parameter_bounds
 
-    def set_as_constraint(self, uncertain_params, **kwargs):
+    def set_as_constraint(self, uncertain_params):
         """
-        Function to generate constraints for the AxisAlignedEllipsoid uncertainty set.
+        Generate constraint(s) for the `AxisAlignedEllipsoidSet`
+        class.
 
         Args:
-           uncertain_params: uncertain parameter objects for writing constraint objects
-        """
-        if len(uncertain_params) != len(self.center):
-               raise AttributeError("Center of ellipsoid must be same dimensions as vector of uncertain parameters.")
-        # square and invert half lengths
-        inverse_squared_half_lengths = list(1.0/(a**2) for a in self.half_lengths)
-        # Calculate row vector of differences
-        diff_squared = []
-        # === Assume VarList uncertain_param_vars
-        for idx, i in enumerate(uncertain_params):
-           if uncertain_params[idx].is_indexed():
-               for index in uncertain_params[idx]:
-                   diff_squared.append((uncertain_params[idx][index] - self.center[idx])**2)
-           else:
-               diff_squared.append((uncertain_params[idx] - self.center[idx])**2)
+            uncertain_params: uncertain parameter objects for writing
+            constraint objects. Indexed parameters are accepted, and
+            are unpacked for constraint generation.
+        """
+        all_params = list()
+
+        # expand all uncertain parameters to a list.
+        # this accounts for the cases in which `uncertain_params`
+        # consists of indexed model components,
+        # or is itself a single indexed component
+        if isinstance(uncertain_params, list):
+            for param in uncertain_params:
+                if param.is_indexed():
+                    all_params.extend(list(param.values()))
+                else:
+                    all_params.append(param)
```

Comment:
Do you need the `is_indexed()` test here?  I think `.values()` should work for both scalar and indexed.  In addition, you don't need the `list(...)`, as `extend()` accepts iterables / generators.  I think the following should work:
```python
if not isinstance(uncertain_params, (tuple, list)):
    uncertain_params = [uncertain_params]
all_params = []
for uparam in uncertain_params:
    all_params.extend(uparam.values())
```

Reply:
See subsequent commits.